Minor API changes

Changaco opened the pull request:
This branch improves the `/%username/public.json` endpoint.

whit537 commented:
What's the reason for this change? I don't see a ticket linked.

I also don't find API doc on https://github.com/liberapay/liberapay.com/ or https://liberapay.com/ to know what's documented and not.

whit537 commented on tests/py/test_public_json.py:
```diff
@@ -70,7 +70,6 @@ def test_jsonp_works(self):
     "avatar": null,
     "elsewhere": {
         "github": {
-            "id": %(elsewhere_id)s,
```
Test the domain addition as well?

Changaco commented:
The reason for this change is that I was looking at `public.json` and it bothered me that the elsewhere data was incomplete, so I fixed it because it was simple to do. Then I realized the `id` keys are useless and confusing so I dropped them.

`public.json` is not documented. The ticket for a proper HTTP API with documentation is #688.

whit537 commented:
> `public.json` is not documented.

In that case I don't see anything to worry about here. 👍
